internal/compiler/output_columns.go

```diff
@@ -764,3 +779,74 @@ func findColumnForRef(ref *ast.ColumnRef, tables []*Table, targetList *ast.List)
 
 	return nil
 }
+
+func convertCaseExprCondToColumn(n ast.Node, resTargetName *string) (*Column, error) {
+	var col *Column
+	name := ""
+	if resTargetName != nil {
+		name = *resTargetName
+	}
+
+	if tc, ok := n.(*ast.TypeCast); ok {
+		if tc.TypeName == nil {
+			return nil, errors.New("no type name type cast")
+		}
+		if ref, ok := tc.Arg.(*ast.ColumnRef); ok {
+			name = astutils.Join(ref.Fields, "_")
+		}
+		// TODO Validate column names
+		col = toColumn(tc.TypeName)
+
+		if x, ok := tc.Arg.(*ast.A_Const); ok {
+			if _, ok := x.Val.(*ast.Null); ok {
+				col.NotNull = false
+			}
+		}
+		col.Name = name
+
+	} else if aconst, ok := n.(*ast.A_Const); ok {
+		col = convertAConstToColumn(aconst, name)
+	} else if aexpr, ok := n.(*ast.A_Expr); ok {
+		col = convertAExprToColumn(aexpr, name)
+	} else {
+		col = &Column{Name: name, DataType: "any", NotNull: false}
+	}
+
+	return col, nil
+}
+
+func convertAExprToColumn(aexpr *ast.A_Expr, name string) *Column {
+	var col *Column
+	switch op := astutils.Join(aexpr.Name, ""); {
+	case lang.IsComparisonOperator(op):
+		// TODO: Generate a name for these operations
+		col = &Column{Name: name, DataType: "bool", NotNull: true}
+	case lang.IsMathematicalOperator(op):
+		col = &Column{Name: name, DataType: "int", NotNull: true}
+	case lang.IsJSONOperator(op) && lang.IsJSONResultAsText(op):
+		col = &Column{Name: name, DataType: "text", NotNull: false}
+	default:
+		col = &Column{Name: name, DataType: "any", NotNull: false}
+	}
+
+	return col
+}
+
+func convertAConstToColumn(aconst *ast.A_Const, name string) (*Column) {
+	var col *Column
+	switch aconst.Val.(type) {
+	case *ast.String:
+		col = &Column{Name: name, DataType: "text", NotNull: true}
+	case *ast.Integer:
+		col = &Column{Name: name, DataType: "int", NotNull: true}
+	case *ast.Float:
+		col = &Column{Name: name, DataType: "float", NotNull: true}
+	case *ast.Boolean:
+		col = &Column{Name: name, DataType: "bool", NotNull: true}
+	case *ast.Null:
```

Comment:
this is new addition to handle null const value. I need to distinguish between `any` and `null`, because `null` in `CaseExpr` would determine query column result nullability, but does not make it included in column type decision.

for `ast.A_Const` condition, i have converted DataType `null` back to `any` (line 136, `outputColumns` function) because we don't know its type and to preserve backward compatibility